src/DependencyInjection/CsaGuzzleExtension.php

```diff
@@ -123,4 +119,16 @@ private function processClientsConfiguration(array $config, ContainerBuilder $co
             }
         }
     }
+
+    private function findSubscriberIds(array $explicitlyConfiguredIds, ContainerBuilder $container)
+    {
+        $allNames = array_values(array_map(
+            function ($tags) { return $tags[0]['alias']; },
+            $container->findTaggedServiceIds(SubscriberPass::SUBSCRIBER_TAG)
+        ));
+
+        return array_filter($allNames, function ($name) use ($explicitlyConfiguredIds) {
+            return !isset($explicitlyConfiguredIds[$name]) || $explicitlyConfiguredIds[$name];
+        });
+    }
```

Comment:
@csarrazi yep, should have done it in the compiler pass. And yep, we can tweak it with subsequent PR-s :)